### ISSUE TITLE
Problem with regex parser including accents when it is inside sequenceOf

### ISSUE THREAD

**Issue by srsergiorodriguez** [francisrstokes/arcsecond#54]
Dear @francisrstokes, first I want to thank you for the work you've done with this library and your Youtube videos, I've learned so much with both.
The problem I'm having is this:
I tried to make a **sequenceOf** parser that includes a **str**, a **regex** (for a string containing any letters, including accent caracters common in Spanish like áéíóú), and then another **str** parser. However, when I run the **sequenceOf** with a string that contains any of those mentioned accents, the parser shows an error in which, apparently, the accented character has been converted to an unknown symbol. When the same problematic **regex** is run alone, the accented character maintains its integrity.
Here's a simplified replication of the problem:

```javascript
const working = regex((/^(\([a-zú]+\))/))
console.log(working.run('(aeioú)'));
// {isError: false, result: "(aeioú)", index: 7, data: null}

const failing = sequenceOf([
    str("("),
    regex(/^([a-zú]+)/),
    str(")")
]);
console.log(failing.run('(aeioú)'));
// {isError: true, error: "ParseError (position 6): Expecting string ')', got '�...'", index: 6, data: null}

const alsoWorking = sequenceOf([
    str("("),
    regex(/^([a-zú]+)/),
    str(")")
]);
console.log(alsoWorking.run('(aeiou)')); // <- string without accented characters
// {isError: false, result: Array(3), index: 7, data: null}
```
The problem repeats if the **str** parsers are replaced by **char** or by **regex** parsers

**Comment by francisrstokes** [comment]
Hey @srsergiorodriguez - I took a look into your issue this morning and it looks like the problem is that the regex parser updates the internal index based on how many characters were matched, instead of how many bytes (`aeioú` has 5 letters but 6 bytes), so the index is still pointing at `ú` when trying to match `)`.

Should be simple enough to fix by using the `TextEncoder` to get the true number of bytes. I'll write a fix for this later today, and check if it's happening anywhere else in the code.

Thanks for the report!

**Comment by francisrstokes** [comment]
@srsergiorodriguez I've made a fix in 3.1.2.

Turns out the issue was also present in the `str` as well. Nice catch!
